service-commands/scripts/bootstrap-solana-program-docker.js

```diff
@@ -0,0 +1,73 @@
+const fs = require('fs')
+const { exec, execSync } = require('child_process')
+
+const solanaProgramsPath = `${process.env.PROTOCOL_DIR}/solana-programs`
+const tmp = `/tmp/solana-deploy`
+
+const writeMetadata = async () => {
+  const gitHash = await new Promise(resolve =>
+    exec(`cd ${solanaProgramsPath} && git rev-parse HEAD`, (_, stdout) =>
+      resolve(stdout)
+    )
+  )
+  fs.writeFileSync(
+    `${tmp}/metadata.json`,
+    JSON.stringify(
+      {
+        gitHash,
+        date: new Date().toISOString()
+      },
+      null,
+      ' '
+    )
+  )
+}
+
+const main = async () => {
+  // Spin up the solana validator from a blank state
+  execSync('A run solana-validator up', { stdio: 'inherit' })
+  // Compile all the solana programs and deploy to the validator
+  execSync('A run solana-programs up', { stdio: 'inherit' })
+
+  // Do some setup...
+  execSync(`rm -rf ${tmp}`, { stdio: 'inherit' })
+  execSync(`mkdir -p ${tmp}`, { stdio: 'inherit' })
+
+  // wait for memory to be written to disk
+  await new Promise(resolve => setTimeout(resolve, 1000 * 60))
```

Comment:
what's this for? comment is a bit unclear to me

Reply:
Oh I think it takes a second for the tx to be written from in memory to the ledger folder that we are copying, so I put a short wait here - actually not sure if entirely needed

Reply:
got it, i don't know if it's needed but definitely can't hurt to have it